direct_db_manager.py

```
@@ -2479,14 +2489,398 @@ def save_user(self):
+        logger.info("Stopping DatabaseWorker thread...")
+        self.is_running = False
+        self.mutex.lock()
+        self.wait_condition.wakeAll()
+        self.mutex.unlock()
+
+        if self.isRunning():
+            self.wait(5000)  # Wait up to 5 seconds for thread to finish
+
+        logger.info("DatabaseWorker thread stopped")
+
+class LoadingOverlay(QWidget):
```

Ruff (0.12.2)

`QWidget` may be undefined, or defined from star imports

(F405)